### PR TITLE
Fix-uk-locale-pluralize-in-old-calculator

### PR DESCRIPTION
**Pull request by fh0enix**
dev ticket
https://github.com/ita-social-projects/ZeroWaste/issues/418

## Code reviewers

- [x] @loqimean

### Second Level Review

- [x] @obniavko 

## Summary of issue

Title: Fix Pluralization in Diaper Calculator Results

Description:
This pull request addresses an issue related to pluralization in the Diaper Calculator results. Previously, the pluralized strings for the number of used and to-be-used diapers were not correctly formatted, leading to inaccuracies in the displayed values. The problem has been resolved by fixing the pluralization logic and updating the corresponding tests to ensure accurate localization.

image
*5673 підгузкИ

image
2562 підгузкИ_________________________4941 підгузОК

Changes:

Corrected pluralization logic for the count of used and to-be-used diapers.
Updated tests to validate the accuracy of pluralized strings in the Diaper Calculator results.
image

Testing:

Added new tests to verify the correct pluralization of diaper counts.
Ran existing test suite to ensure overall functionality.
image

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by obniavko — APPROVED** [review]

**Review by obniavko — APPROVED** [review]

File: app/javascript/controllers/results_controller.js [hunk 1]
```diff
@@ -6,14 +6,19 @@ export default class extends Controller {
...
     "moneyWillBeSpent",
+    "willBuyDiapersPluralize",
+    "boughtDiapersPluralize",
   ];
 
   showResults(data) {
     let result = data.result;
 
-    this.moneySpentTarget.innerHTML = Math.ceil(result.money_spent);
+    this.moneySpentTarget.innerHTML = Math.ceil(result.money_spent);;
```
Comment (loqimean) [inline app/javascript/controllers/results_controller.js:16]:
```suggestion
    this.moneySpentTarget.innerHTML = Math.ceil(result.money_spent);
```

File: app/views/calculators/old_calculator.html.slim [hunk 2]
```diff
@@ -35,18 +35,18 @@
                       class: 'calculate-btn result-btn calc-item old-btn',
                       data: { action: 'input-child-info#submit' }
 
-.container.calculation-results.result [data-controller='results']
+.container.calculation-results.result[data-controller='results']
```
Comment (loqimean) [inline app/views/calculators/old_calculator.html.slim:38]: а пробіл точно не треба?

File: spec/requests/diaper_calculators_spec.rb [hunk 3]
```diff
@@ -96,5 +98,19 @@
...
     end
+
+    context "when get awaited values" do
+      include_context :app_config_load
+
+      it "translates used and to-be-used diapers amounts" do
+        post api_v1_diaper_calculators_path, params: { childs_years: 1, childs_months: 0 }
+
+        expected_used_diapers       = I18n.t("calculators.old_calculator.bought_diapers", count: values[:used_diapers_amount])
+        expected_to_be_used_diapers = I18n.t("calculators.old_calculator.will_buy_diapers", count: values[:to_be_used_diapers_amount])
```
Comment (loqimean) [inline spec/requests/diaper_calculators_spec.rb:109]: сподіваюся я такого більше не побачу, для того є така штука, як `let`

**Review by loqimean — CHANGES_REQUESTED** [review]

File: spec/requests/diaper_calculators_spec.rb (hunk 3 above)
Comment (fh0enix) [reply spec/requests/diaper_calculators_spec.rb:109]: solved

File: app/views/calculators/old_calculator.html.slim (hunk 2 above)
Comment (fh0enix) [reply app/views/calculators/old_calculator.html.slim:38]: ou sorry. solved

File: app/javascript/controllers/results_controller.js (hunk 1 above)
Comment (fh0enix) [reply app/javascript/controllers/results_controller.js:16]: oops... solved

File: spec/requests/diaper_calculators_spec.rb [hunk 4]
```diff
@@ -96,5 +98,29 @@
...
+      let(:expected_used_diapers) do
+        I18n.t("calculators.old_calculator.bought_diapers", count: used_diapers_count)
+      end
+
+      let(:expected_to_be_used_diapers) do
+        I18n.t("calculators.old_calculator.will_buy_diapers", count: to_be_used_diapers_count)
+      end
+
+      it "translates used and to-be-used diapers amounts" do
+        post api_v1_diaper_calculators_path, params: { childs_years: 1, childs_months: 0 }
```
Comment (loqimean) [inline spec/requests/diaper_calculators_spec.rb:117]: це в before

File: spec/requests/diaper_calculators_spec.rb [hunk 5]
```diff
@@ -96,5 +98,29 @@
...
+
+        response_body = JSON.parse(response.body)["result"]
```
Comment (loqimean) [inline spec/requests/diaper_calculators_spec.rb:119]: а це в let

**Review by loqimean — CHANGES_REQUESTED** [review]

File: spec/requests/diaper_calculators_spec.rb (hunk 5 above)
Comment (fh0enix) [reply spec/requests/diaper_calculators_spec.rb]: solved

File: spec/requests/diaper_calculators_spec.rb (hunk 4 above)
Comment (fh0enix) [reply spec/requests/diaper_calculators_spec.rb]: solved

**Pull request merged**